include/deal.II/lac/tensor_product_matrix.h

```diff
@@ -116,11 +116,27 @@ class TensorProductMatrixSymmetricSumBase
    * described in the main documentation of TensorProductMatrixSymmetricSum.
    * This function is operating on ArrayView to allow checks of
    * array bounds with respect to @p dst and @p src.
+   *
+   * @warning This function works on an internal temporal array, leading to
+   * increased memory consumption if many instances of this class is created,
+   * e.g., in the case of non-Cartesian meshes. Futhermore, only one thread run
+   * this function at once (ensured internally with a mutex). If these two
+   * limitations are an issue for you, please consider the other version of this
+   * function.
    */
   void
   apply_inverse(const ArrayView<Number> &      dst,
                 const ArrayView<const Number> &src) const;
 
+  /**
+   * Same as above but the user can provide a user-owned temporal array,
+   * resolving the two issues described above.
+   */
```

Comment:
It looks like the temporary array needs to be of length `2 * n`: could you document that here?

Reply:
The array is set to the right size internally.

Reply:
Got it. Could we document that instead? We typically require that users set output arrays to the right sizes so we should clarify what to do for work arrays.

Reply:
I have added a comment!